ambry-rest/src/test/java/com.github.ambry.rest/NettyRequestTest.java

```diff
@@ -653,6 +660,11 @@ private void validateRequest(NettyRequest nettyRequest, RestMethod restMethod, S
     assertEquals("Mismatch in uri", uri, nettyRequest.getUri());
     assertNotNull("There should have been a RestRequestMetricsTracker", nettyRequest.getMetricsTracker());
     assertFalse("Should not have been a multipart request", nettyRequest.isMultipart());
+    if (channel.pipeline().get(SslHandler.class) == null) {
+      assertNull("Non-null SSLSession when pipeline does not contain an SslHandler", nettyRequest.getSSLSession());
+    } else {
+      assertNotNull("Null SSLSession when pipeline contains an SslHandler", nettyRequest.getSSLSession());
```

Comment:
Because no request has gone through the handler, the certificates in the session are null, but I can check for equality with the expected `SSLSession` object.